include/sway/tree/view.h

```diff
@@ -29,15 +29,15 @@ struct sway_view_impl {
 	const char *(*get_string_prop)(struct sway_view *view,
 			enum sway_view_prop prop);
 	uint32_t (*get_int_prop)(struct sway_view *view, enum sway_view_prop prop);
-	void (*configure)(struct sway_view *view, double lx, double ly, int width,
-		int height);
+	uint32_t (*configure)(struct sway_view *view, double lx, double ly,
+			int width, int height);
 	void (*set_activated)(struct sway_view *view, bool activated);
 	void (*set_fullscreen)(struct sway_view *view, bool fullscreen);
 	bool (*wants_floating)(struct sway_view *view);
 	void (*for_each_surface)(struct sway_view *view,
 		wlr_surface_iterator_func_t iterator, void *user_data);
 	void (*close)(struct sway_view *view);
-	void (*destroy)(struct sway_view *view);
+	void (*free)(struct sway_view *view);
```

Comment:
(nitpick) Even if the implementation has been refactored to separate the rest of the destroy (now in surface notify destroy) and this function that is literally a free with extra checks, I'd keep the name `destroy` here as that's what we have all around in the impl vectors.
You never know if in the future that'll need to do a bit more work again, and it won't be the first destroy function that just frees its argument :)

Reply:
I renamed it to `free` to keep it consistent with the `view_destroy` and `view_free` functions, and `container_destroy` and `container_free`. Destroying and freeing are now two different things. Destroying basically marks it as destroying, and when it's no longer in a transaction it gets freed.

When a surface wants to destroy itself, we must unregister the wayland events before returning from `handle_destroy` because after we return the surface is no longer valid. But we can't free the view until later, when the view is no longer in a transaction. The non-atomic code in master does the unregistering and freeing of the view all in the `destroy` callback, using a call stack of `handle_destroy -> view_destroy -> view->destroy()`.

My branch takes a shortcut by unregistering the events in `handle_destroy`, and `view_destroy` doesn't have to dive back into the implementation at all. I could add the `destroy` callback back and call it from `view_destroy`, but it needs to be separate from the `free` callback. The only benefit of doing it that way is the compositor could theoretically initiate a destroy without having the view requesting it. I'm not sure if that's against the protocol though, or if we'd ever use it.

Reply:
Yeah, I also think a `destroy` function which doesn't frees is confusing. I'd rather keep destroy and pick another name for the other function.